update roles doc (#282)
hats-finance/hats-contracts: docs/roles.md

# Security Analysis HATVaults 2.0

The contracts in this repository define a number of different roles that have the power to change the way the contracts work. We list them here, with a short description of what an account with this role can do, and to which account this role will be assigned on deployment

## `HatVaultsRegistry.owner`

- can call `setFeeSetter` set the feeSetter role
- can call `setArbitrator` set the arbitrator role
- can call `setWithdrawRequestParams` set time limits for withdrawal requests

> - can call `setClaimFee` set fee for submitting a vulnerability

Reviewer: ```suggestion
- can call `setClaimFee` to set the fee for logging a claim for a bounty in any vault.
```

> - can call `setChallengePeriod` 

Reviewer: ```suggestion
- can call `setChallengePeriod` to set the time during which a claim can be challenged by the arbitrator
```

- can call `setChallengeTimeOutPeriod` 

> - can call `setWithdrawSafetyPeriod` set the amount of time during which claims can be submitted; during this time the vault users can not withdraw their funds. Must be less than 6 hours

Reviewer: ```suggestion
- can call `setWithdrawSafetyPeriod` to set the withdraw period and safety period, which are always interchanging. The safety period is time can submit claims for bounty payouts, and during which withdrawals are disabled and the bounty split cannot be changed.
```

Reviewer: more accurate because it sets both

- can call `setHatVestingParams` set vesting paramaters for bounty paid in hats
- `setMaxBountyDelay` - Set the timelock delay for setting the max bounty
- can call `setRouterWhitelistStatus` add or remove address from the whitelist of routers that can be used for token swapping

> - can call `swapBurnSend` and swap, burn, and send the vauit's token that are earmarked for payout (after approveClaim)

Author: why?

> - can call `swapBurnSend` and swap, burn, and send the vauit's token that are earmarked for payout (after approveClaim)

Reviewer: because swapBurnSend is run by the owner of the vault and not the registry

# Security Analysis HATVaults 2.0

**Some of the functions have changed after this document has written**

The contracts in this repository define a number of different roles that have the power to change the way the contracts work. We list them here, with a short description of what an account with this role can do, and to which account this role will be assigned on deployment

## `HatVaults.ProxyOwner`
## `HatVaultsRegistry.owner`

- set to `HATTimelockController`
- can upgrade the proxy

> - can `transferOwnership` and `renounceOwnership`

Reviewer 2: Well that is possible no?

- set to governance multisig
- can call `cancel` and cancel any pending operation

> 

Reviewer: ```suggestion
- can call `setWithdrawalFee` on any vault - set the fee for withdrawals from the vault

```

## The following functions in HATVaults are **not** subject to a timelock:
  - `approveClaim`

  - `setPool`
  - `updateVaultInfo`

>   - `setAllocPoint`

Reviewer: ```suggestion
  - `setAllocPoint`
  - `setCommittee`
```
